### PR TITLE
TypeScript 5: Convert schemas

### PR DESCRIPTION
**Pull request by alukach**
## What I'm changing

This PR converts all of our Mongoose schema files to TS.

## How I did it

The conversion was pretty straight forward:

1. Rename all files from *.js to *.ts
2. Add a typing for the untyped `mongo-cursor-pagination` module
3. While I was at it, I ran `prettier` on all of the files as well.

### UPDATE

I've updated this PR with [0022ca2](https://github.com/tnc-ca-geo/animl-api/pull/205/commits/0022ca273db3b5aa77e57dfc954b6e7c9be29c58) to export a schema type from every file.  This will provide us a type to conveniently describe the output of DB queries, such as:

```ts
import Batch, { BatchSchema } from '../schema/batch.js'

function query(): Promise<BatchSchema> {
    return Batch.findOne()
}
```

---

Relates to #187

### CODE REVIEW

File: src/api/db/schemas/Batch.ts [hunk 1]
```diff
@@ -4,19 +4,21 @@ import MongoPaging from 'mongo-cursor-pagination';
...
   user: { type: String },
   created: { type: Date, default: Date.now },
   overrideSerial: { type: String },
   uploadedFile: { type: String },
   originalFile: { type: String },
-  uploadComplete: { type: Date }, /* Step 1 complete - Zip file has been fully received and validated */
-  processingStart: { type: Date }, /* Step 2 complete - Cloudformation stack has been deployed */
-  ingestionComplete: { type: Date }, /* Step 3 complete - All files have been saved to DB */
-  processingEnd: { type: Date }, /* Step 4 complete - All files have been saved to DB */
+  uploadComplete: {
```
Comment (jue-henry) [inline src/api/db/schemas/Batch.ts:14]: Was this just changed to multiline because of line length?

File: src/@types/mongo-cursor-pagination.d.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  import { Collection } from 'mongoose';
+
+  export type AggregationInput<T = {}> = T & {
+    paginatedField: string;
+    sortAscending: boolean;
+    limit: number;
+    next: string;
+    previous: string;
+  };
+  export interface AggregationOutput<T> {
```
Comment (jue-henry) [inline src/@types/mongo-cursor-pagination.d.ts:11]: nitpick but newline above?

**Review by jue-henry — APPROVED** [review]

File: src/api/db/schemas/Batch.ts (hunk 1 above)
Comment (alukach) [reply src/api/db/schemas/Batch.ts:14]: Yeah, just `prettier` doing its thing.

**Review by nathanielrindlaub — APPROVED** [review]
Looks great to me @alukach. Merging :)

**Pull request merged**